theme_selector: Don't select last theme when fuzzy searching

hoodie opened the pull request:
The theme selector should select the last match to when opening to keep the current theme active but it should select the first when searching to pick the best match.

fixes #28081

https://github.com/user-attachments/assets/b46b9742-4715-4c7a-8f17-2c19a8668333

Release Notes:

- Fixed selecting the correct theme when searching

hoodie committed:
Fix select Theme Dialog selects last theme when fuzzy searching

The theme selector should select the last match to when opening to keep
the current theme active but it should select the first when searching
to pick the best match.

fixes #28081

notpeter commented:
Tested and it definitely feels better when searching to have the selected match be the top, but it also means if you have done any selection (up/down) and then type or hit backspace it loses your selection even when that would still be the correct match. Ideally I'd like to preserve that behavior and fix the bottom selection too. 

https://github.com/user-attachments/assets/38dbee2b-fbf1-486b-9c9b-7462f0280548

hoodie commented:
True. Incidently this is also the behavior of in vs-code.

https://github.com/user-attachments/assets/2f0c586c-1dd1-478e-a75c-02dec9334078


If you want to avoid this you'd have to switch to a different mode once changing selection index via up/down keys right?

hoodie commented:
I think I got something

hoodie committed:
Add manual theme selection persistence

Store manually selected themes to maintain selection when query changes.
This allows the theme selector to remember the user's choice rather than
resetting selection when filtering results.

hoodie commented:
https://github.com/user-attachments/assets/2eca3379-2c51-43eb-8334-32200b08a081
I'm persisting the selected theme when selecting via up/down and if the query changes I search the selected theme inside them matches

hoodie commented:
I just realized that the above video shows another issue. The current theme matches what I type and should probably stay selected. Now I persist the selected theme better.

https://github.com/user-attachments/assets/535240f7-4058-492c-ba1e-a505a724c0af

hoodie commented:
No worries. I think this change can probably be applied to at least the icon theme picker as well.

hoodie commented:
I did that already [here](https://github.com/hoodie/zed/tree/bugfix/select-correct-icon-theme-when-searching), can open a PR if you want

notpeter committed:
Merge branch 'main' into bugfix/select-correct-theme-when-searching

notpeter commented:
Tested and this seems to be an improvement. Thanks!
Feel free to open a PR with changes to match this behavior for the icon theme selector. 👯

hoodie commented:
woop woop, thanks. will do
